system-addon/common/Reducers.jsm

```diff
@@ -97,16 +97,43 @@ function TopSites(prevState = INITIAL_STATE.TopSites, action) {
       return Object.assign({}, prevState, {
         editForm: {
           index: prevState.editForm.index,
-          screenshotRequestFailed: true
+          screenshotRequestFailed: true,
+          requestedScreenshotURL: null
```

Comment:
Why do we need to clear the requestedScreenshotURL? I would have expected this FAILED action to check if the failing preview was actually for the screenshot being requested. Similar to the check for SCREENSHOT_PREVIEW.

Reply:
As long as `requestedScreenshotURL` is set, we show the loading/pending state. Added the check as mentioned.